lib/rust/parser/src/macros/built_in.rs

```diff
@@ -401,25 +401,20 @@ fn try_parse_doc_comment<'s>(
     items: &mut Vec<Item<'s>>,
     precedence: &mut operator::Precedence<'s>,
 ) -> Option<syntax::tree::DocComment<'s>> {
-    if let Some(Item::Token(token @ Token { variant: token::Variant::TextStart(_), .. })) =
-        items.first()
-        && token.code == "##"
-    {
+    if matches!(
+        items.first(),
+        Some(Item::Token(token @ Token { variant: token::Variant::TextStart(_), .. })) if token.code == "##"
+    ) {
         let Some(syntax::Tree {
-            variant:
-                syntax::tree::Variant::Documented(box syntax::tree::Documented {
-                    mut documentation,
-                    expression: None,
-                    ..
-                }),
+            variant: syntax::tree::Variant::Documented(mut documented),
```

Comment:
I think we lost the check if `expression` is None.

Reply:
The alternative is unreachable, so I'm not sure it matters in this case. @kazcw could you check if that match was indeed significant here?